File: Java/OPSJLib/src/ops/DefaultOPSConfigImpl.java
/**
*
* Copyright (C) 2006-2009 Anton Gravestam.
* Copyright (C) 2020 Lennart Andersson.
*
* This file is part of OPS (Open Publish Subscribe).
*
* OPS (Open Publish Subscribe) is free software: you can redistribute it and/or modify
* it under the terms of the GNU Lesser General Public License as published by
* the Free Software Foundation, either version 3 of the License, or
* (at your option) any later version.

* OPS (Open Publish Subscribe) is distributed in the hope that it will be useful,
* but WITHOUT ANY WARRANTY; without even the implied warranty of
* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
* GNU Lesser General Public License for more details.
*
* You should have received a copy of the GNU Lesser General Public License
* along with OPS (Open Publish Subscribe).  If not, see <http://www.gnu.org/licenses/>.
*/

package ops;

import configlib.ArchiverInOut;
import java.io.IOException;

/**
 *
 * @author angr
 */
public class DefaultOPSConfigImpl extends OPSConfig
{
    public byte DefaultOPSConfigImpl_version = DefaultOPSConfigImpl_idlVersion;

    public static final byte DefaultOPSConfigImpl_idlVersion = 0;

    public DefaultOPSConfigImpl()
    {
        appendType("DefaultOPSConfigImpl");
    }

    @Override
    public void serialize(ArchiverInOut archive) throws IOException
    {
        super.serialize(archive);
        if (idlVersionMask != 0) {
            byte tmp = archive.inout("DefaultOPSConfigImpl_version", DefaultOPSConfigImpl_version);
            if (tmp > DefaultOPSConfigImpl_idlVersion) {
                throw new IOException("DefaultOPSConfigImpl: received version '" + tmp + "' > known version '" + DefaultOPSConfigImpl_idlVersion + "'");
            }
            DefaultOPSConfigImpl_version = tmp;
        } else {
            DefaultOPSConfigImpl_version = 0;
        }
    }

}

File: CSharp/Ops/DefaultOPSConfigImpl.cs
///////////////////////////////////////////////////////////
//  DefaultOPSConfigImpl.cs
//  Implementation of the Class DefaultOPSConfigImpl
//  Created on:      12-nov-2011 09:25:29
//  Author:
///////////////////////////////////////////////////////////

namespace Ops 
{
	public class DefaultOPSConfigImpl : OPSConfig 
    {
        private byte _DefaultOPSConfigImpl_version = DefaultOPSConfigImpl_idlVersion;
        public byte DefaultOPSConfigImpl_version
        {
            get { return _DefaultOPSConfigImpl_version; }
            set
            {
                if (value > DefaultOPSConfigImpl_idlVersion)
                {
                    throw new IdlVersionException(
                        "DefaultOPSConfigImpl: received version '" + value + "' > known version '" + DefaultOPSConfigImpl_idlVersion + "'");
                }
                _DefaultOPSConfigImpl_version = value;
            }
        }

        public const byte DefaultOPSConfigImpl_idlVersion = 0;

        public DefaultOPSConfigImpl()
        {
            AppendType("DefaultOPSConfigImpl");
        }

        public override void Serialize(IArchiverInOut archive) 
        {
            base.Serialize(archive);
            if (IdlVersionMask != 0)
            {
                DefaultOPSConfigImpl_version = archive.Inout("DefaultOPSConfigImpl_version", DefaultOPSConfigImpl_version);
            }
            else
            {
                DefaultOPSConfigImpl_version = 0;
            }
        }
    }

}